File: adapter/java/concrete/AudioPlayer.java
package adapter.java.concrete;

import adapter.java.interfaces.MediaPlayer;
import adapter.java.adapter.MediaAdapter;


// Cliente que usa el adaptador
public class AudioPlayer implements MediaPlayer{
    @Override
    public void play(String audioType, String fileName) {
        // Reproducir archivos mp3
        if(audioType.equalsIgnoreCase("mp3")){
            System.out.println("Reproduciendo archivo mp3. Nombre: " + fileName);
        } 
        // Reproducir archivos mp4 y vlc
        else if(audioType.equalsIgnoreCase("mp4") || audioType.equalsIgnoreCase("vlc")){
            MediaAdapter mediaAdapter = new MediaAdapter(audioType);
            mediaAdapter.play(audioType, fileName);
        } 
        // No se puede reproducir el archivo
        else{
            System.out.println("Formato de archivo no soportado: " + audioType);
        }
    }
}

File: adapter/php/concrete/AudioPlayer.php
<?php
namespace Concrete;

require_once 'legacy/AdvancedMediaPlayer.php';
require_once 'adapter/MediaAdapter.php';

use Interfaces\MediaPlayer;
use Adapter\MediaAdapter;

// Cliente que usa el adaptador
class AudioPlayer implements MediaPlayer{
    public function play($audioType, $fileName)
    {
        // Reproductor nativo soporta mp3
        if ($audioType == "mp3") {
            echo "Reproduciendo archivo mp3. Nombre: " . $fileName . "\n";
        } else if ($audioType == "vlc" || $audioType == "mp4") {
            $mediaAdapter = new MediaAdapter($audioType);
            $mediaAdapter->play($audioType, $fileName);
        } else {
            echo "Formato de audio no soportado\n";
        }
    }
}